pipenv/cli.py

```diff
@@ -105,11 +105,16 @@ def ensure_pipfile(validate=True):
             # Pip requires a `PipSession` which is a subclass of requests.Session.
             # Since we're not making any network calls, it's initialized to nothing.
             from pip.req.req_file import parse_requirements
-            reqs = [r.req for r in parse_requirements(project.requirements_location, session='')]
+            reqs = [r for r in parse_requirements(project.requirements_location, session='')]
 
             for package in reqs:
                 if package.name not in BAD_PACKAGES:
-                    project.add_package_to_pipfile(str(package))
+                    if package.original_link is not None:
+                        pack_link = str(package.original_link)
```

Comment:
Yes, the intention was to call a str magic method on a requirement object, although original link might always be a string, I put it in there with some other changes and it was working, so this could be removed, I'll test.